lib/elliptic/ec/index.js

```diff
@@ -164,3 +169,47 @@ EC.prototype.verify = function verify(msg, signature, key, enc) {
 
   return p.getX().mod(this.n).cmp(r) === 0;
 };
+
+EC.prototype.recoverPubKey = function(msg, signature, j) {
+  j = new bn(j);
+  var n = this.n;
+  var e = new bn(msg);
+  var r = signature.r;
+  var s = signature.s;
+
+  // A set LSB signifies that the y-coordinate is odd
+  var isYOdd = j & 1;
+
+  // The more significant bit specifies whether we should use the
+  // first or second candidate key.
+  var isSecondKey = j >> 1;
```

Comment:
I think it is pretty odd to do the regular arithmetic ops on BN instances. What do you think about doing `j.shrn(1)`?